src/platform/forms-system/src/js/web-component-patterns/addressPattern.jsx

```diff
@@ -0,0 +1,479 @@
+        ? savedAddress.stateCode
+        : stateCode || savedAddress.stateCode;
+    }
+    // make sure we aren't splitting up a string path
+    const pathArray = Array.isArray(path) ? path : [path];
+    updatedData = set([...pathArray, schemaKeys.city], city, updatedData);
+    updatedData = set([...pathArray, schemaKeys.state], stateCode, updatedData);
+  }
+  return updatedData;
+};
+
+/**
+ * @typedef {'country' | 'city' | 'isMilitary' | 'postalCode' | 'state' | 'street' | 'street2' | 'street3' } AddressSchemaKey
+ */
+
+/**
+ * Web component uiSchema for address
+ *
+ * ```js
+ * schema: {
+ *   address: addressUI()
+ *   simpleAddress: addressUI({ omit: ['street2', 'street3'] })
+ *   futureAddress: addressUI({ militaryCheckboxLabel: 'I will live on a United States military base outside of the U.S.'; })
+ *   changeRequired: addressUI({
+ *     required: {
+ *       country: (formData) => false,
+ *       street2: (formData) => true
+ *     }
+ *   })
+ * }
+ * ```
+ * @param {{
+ *   militaryCheckboxLabel?: string,
+ *   omit?: Array<AddressSchemaKey>,
+ *   required?: Record<AddressSchemaKey, (formData:any) => boolean>
+ * }} [options]
+ * @returns {UISchemaOptions}
+ */
+export function addressUI(options) {
+  let cachedPath;
+  const omit = key => options?.omit?.includes(key);
+  const customRequired = key => options?.required?.[key];
+
+  /** @type {UISchemaOptions} */
+  const uiSchema = {};
+
+  if (!omit('isMilitary')) {
+    uiSchema.isMilitary = {
+      'ui:title':
+        options?.militaryCheckboxLabel ??
+        'I live on a United States military base outside of the U.S.',
+      'ui:webComponentField': VaCheckboxField,
+      'ui:options': {
+        hideEmptyValueInReview: true,
+      },
+      'ui:required': customRequired('isMilitary') || (() => false),
+    };
+  }
+
+  if (!omit('isMilitary') && !omit('view:militaryBaseDescription')) {
+    uiSchema['view:militaryBaseDescription'] = {
+      'ui:description': MilitaryBaseInfo,
+    };
+  }
+
+  if (!omit('country')) {
+    uiSchema.country = {
+      'ui:required': formData => {
+        if (customRequired('country')) {
+          return customRequired('country')(formData);
+        }
+        if (cachedPath) {
+          const { isMilitary } = get(cachedPath, formData) ?? {};
+          return !isMilitary;
+        }
+        return true;
+      },
+      'ui:title': 'Country',
+      'ui:autocomplete': 'country',
+      'ui:webComponentField': VaSelectField,
+      'ui:errorMessages': {
+        required: 'Country is required',
+      },
+      'ui:options': {
+        /**
+         * This is needed because the country dropdown needs to be set to USA and disabled if a
+         * user selects that they live on a military base outside the US.
+         */
+        updateSchema: (formData, schema, _uiSchema, index, path) => {
+          const addressPath = path.shift(); // path is ['address', 'currentField']
```

Comment:
before we had to pass the path into address, but I discovered there is a 5th property here that includes it for us. I'm not sure if it will work properly with arrays yet. there was logic before that allowed you to pass [:INDEX] in the string.